File: engine/model/src/main/scala/tools/packet/character/interaction/MerchantChat.scala
package tools.packet.character.interaction

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MerchantChat(private var _message: String, private var _slot: Byte) extends PacketInput {
  def message: String = _message

  def slot: Byte = _slot

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/MerchantChat.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MerchantChat(String message, Byte slot) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}